### ISSUE TITLE
Compiling issue when adding module

### ISSUE THREAD

**Issue by Jiahao-Liu-Jeff** [stnoonan/spnego-http-auth-nginx-module#127]
Here are the error details.

../spnego-http-auth-nginx-module/ngx_http_auth_spnego_module.c:761:9: error: variable 'kcontext' is used uninitialized whenever 'if' condition is true [-Werror,-Wsometimes-uninitialized]
    if (!delegated_creds.data) {
        ^~~~~~~~~~~~~~~~~~~~~
../spnego-http-auth-nginx-module/ngx_http_auth_spnego_module.c:838:9: note: uninitialized use occurs here
    if (kcontext)
        ^~~~~~~~
../spnego-http-auth-nginx-module/ngx_http_auth_spnego_module.c:761:5: note: remove the 'if' if its condition is always false
    if (!delegated_creds.data) {
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../spnego-http-auth-nginx-module/ngx_http_auth_spnego_module.c:754:26: note: initialize the variable 'kcontext' to silence this warning
    krb5_context kcontext;
                         ^
                          = NULL
../spnego-http-auth-nginx-module/ngx_http_auth_spnego_module.c:1531:28: error: implicit declaration of function 'gss_localname' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
            major_status = gss_localname(&minor_status, client_name, mech_type,
                           ^
2 errors generated.
make[1]: *** [objs/addon/spnego-http-auth-nginx-module/ngx_http_auth_spnego_module.o] Error 1
make: *** [modules] Error 2

**Comment by oranki** [comment]
I believe this commit fixes the warnings:
https://github.com/arduent/spnego-http-auth-nginx-module/commit/c58216b18af63a52d6bd9cf29df4d695c7da07e3

**Comment by Jiahao-Liu-Jeff** [comment]
> I believe this commit fixes the warnings: [arduent@c58216b](https://github.com/arduent/spnego-http-auth-nginx-module/commit/c58216b18af63a52d6bd9cf29df4d695c7da07e3)

Right, this commit fixes the variable warning but compiling error of 'gss_localname' still exist. Thank you for you reply though.

**Comment by oranki** [comment]
gss_localname() is included from <gssapi/gssapi_ext.h>, via <gssapi/gssapi_krb5.h>

**Comment by Jiahao-Liu-Jeff** [comment]
> gss_localname() is included from <gssapi/gssapi_ext.h>, via <gssapi/gssapi_krb5.h>

I tried to compile this module at MacOS and relative gss_localname() error occurs. But it seems to be right when I used same command in CentOS docker container. I don't know if the environment matters.

**Comment by oranki** [comment]
`gss_localname()` is defined in gssapi headers. Since you're not getting errors from missing headers, it seems likely your version of the headers do not have `gss_localname()` function. You probably need to either update them somehow, or install another Kerberos library to link against.

**Comment by Jiahao-Liu-Jeff** [comment]
> `gss_localname()` is defined in gssapi headers. Since you're not getting errors from missing headers, it seems likely your version of the headers do not have `gss_localname()` function. You probably need to either update them somehow, or install another Kerberos library to link against.

Thank you for you reply. Now I am working on resolving web browser access to page which protected by Kerberos. I configured Nginx with spnege module and set the conf file like below:

'
llocation / {
    auth_gss on;
    auth_gss_realm EXAMPLE.ORG;
    auth_gss_service_name "HTTP/test.example.org";
    auth_gss_keytab /etc/krb5.keytab;
   return 301 xxxxx;
  }
'
But it still return 403. Is it possible to access protected page like this way?

**Comment by Alphix** [comment]
Can't see anything actionable here, closing
